lib/model-interfaces/bedrock-agents/functions/request-handler/index.py

```diff
@@ -0,0 +1,163 @@
+                        "metadata": {
+                            "modelId": agent_id,
+                            "modelKwargs": None,
+                            "mode": "agent",
+                            "sessionId": session_id,
+                            "userId": user_id,
+                            "documents": [],
+                            "prompts": [prompt],
+                        },
+                    },
+                }
+            )
+        if "trace" in list(r.keys()):
+            send_to_client(
+                {
+                    "type": "text",
+                    "action": ChatbotAction.AGENT_TRACE.value,
+                    "timestamp": str(int(round(datetime.now().timestamp()))),
+                    "userId": user_id,
+                    "data": {
+                        "sessionId": session_id,
+                        "type": "text",
+                        "sequence": sequence_number,
+                        "runId": run_id,
+                        "content": json.dumps(r["trace"]),
+                        "metadata": {
+                            "modelId": agent_id,
+                            "modelKwargs": None,
+                            "mode": "agent",
+                        },
+                    },
+                }
+            )
+            sequence_number += 1
+
+
+@tracer.capture_method
+def record_handler(record: SQSRecord):
+    payload: str = record.body
+    message: dict = json.loads(payload)
+    detail: dict = json.loads(message["Message"])
+    logger.info(detail)
+
+    if detail["action"] == ChatbotAction.RUN.value:
+        handle_run(detail)
+    elif detail["action"] == ChatbotAction.HEARTBEAT.value:
+        pass
+
+
+def handle_failed_records(records):
+    for triplet in records:
+        _, error, record = triplet
+        payload: str = record.body
+        message: dict = json.loads(payload)
+        detail: dict = json.loads(message["Message"])
+        logger.info(detail)
+        user_id = detail["userId"]
+        data = detail.get("data", {})
+        session_id = data.get("sessionId", "")
+
+        send_to_client(
+            {
+                "type": "text",
+                "action": "error",
+                "direction": "OUT",
+                "userId": user_id,
+                "timestamp": str(int(round(datetime.now().timestamp()))),
+                "data": {
+                    "sessionId": session_id,
+                    "content": str(error),
+                    "type": "text",
+                },
+            }
+        )
+
+
+@logger.inject_lambda_context(log_event=True)
+@tracer.capture_lambda_handler
+def handler(event, context: LambdaContext):
+    batch = event["Records"]
+
+    try:
+        with processor(records=batch, handler=record_handler):
+            processed_messages = processor.process()
+    except BatchProcessingError as e:
+        logger.error(e)
+
+    logger.info(processed_messages)
```

Comment:
```suggestion
for message in processed_messages:
        logger.info(
            "Request complete with status " + message[0],
            status=message[0],
            cause=message[1],
        )
```

The problem is message[2] is the raw exception that might contain sensitive information like headers.